File: hastaneSistemi/src/app/services/working-hours.service.ts
// src/app/services/working-hours.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { WorkingHours } from '../models/working-hours.model';

@Injectable({
  providedIn: 'root'
})
export class WorkingHoursService {
  private apiUrl = 'http://localhost:8080/api/working-hours'; // Base URL

  constructor(private http: HttpClient) {}

  getWorkingHoursByDoctor(doctorId: number): Observable<WorkingHours[]> {
    return this.http.get<WorkingHours[]>(`${this.apiUrl}/doctor/${doctorId}`);
  }
  addWorkingHours(workingHours: WorkingHours): Observable<WorkingHours> {
    return this.http.post<WorkingHours>(this.apiUrl, workingHours);
  }

  deleteWorkingHours(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  updateWorkingHours(id: number, workingHour: WorkingHours) {
    return this.http.put(`${this.apiUrl}/${id}`, workingHour);
  }
  
  
}

File: HYBS backend/src/main/java/com/example/randevu/service/WorkingHoursService.java
package com.example.randevu.service;

import com.example.randevu.entity.WorkingHours;

import java.util.List;

public interface WorkingHoursService {
    WorkingHours saveWorkingHours(WorkingHours workingHours);
    WorkingHours updateWorkingHours(Long id, WorkingHours workingHours);
    WorkingHours getWorkingHoursById(Long id);
    List<WorkingHours> getWorkingHoursByDoctorId(Long doctorId);
    void deleteWorkingHours(Long id);
    List<WorkingHours> getDoctorWorkingHours(Long doctorId);
}
